Abstract PML definitions from YeePML field solver

steindev opened the pull request:
PMLs could be used in other field solvers, too. But to do so, the variables of the PML in `pml.param` need to be separated from the `YeePML` field solver. This PR puts the definitions of the PML variables in their own namespace `PML` and updates the `YeePML` field solver to use the variables from this namespace.

A test with laser propagation in vacuum using PML boundaries compiles without error. (I am not sure if there is one compile test in the CI using PMLs.)

sbastrakov commented on include/picongpu/param/pml.param:
```diff
@@ -33,11 +33,11 @@ namespace fields
 {
 
     /* The parameters in this file are only used if the field solver selected is
-     * YeePML.
+     * uses Perfectly Matched Layer Absorbing Boundary Conditions (PML).
```
I guess need to remove "is" on line 39.

steindev replied:
Yes, thanks. Removed it.

steindev commented:
I rebased my branch and squashed my commits. Now PrometheusPi 's latest commit to the dev branch `add verbose output for openPMD-api` shows up in my history as well, see above. Is this normal, or should I do something about it?

sbastrakov commented:
@steindev I think this is because I used the wrong merge option for that PR of his. (Changed to squash for your other PR, and then changed "back" to the wrong one). To be sure, you drop that commit from your branch and then rebase on the current `dev`. (I guess there are other ways to achieve that as well, I would have just done it like that)

psychocoderHPC commented:
note: ci/gitlab/gitlab.com can be ignored, the CI is not required for a merge

steindev commented:
> @steindev I think this may be because I used the wrong merge option for that PR of his. (Changed to squash for your other PR, and then changed "back" to the wrong one). To be sure, you could drop that commit from your branch and then rebase your only real commit on the current `dev`. (I guess there are other ways to achieve that as well, I would have just done it like that)

Thanks that worked. I think, I also did a mistake by first rebasing and then squashing. As far as I have seen now, the other way round seems to be the common workflow.

sbastrakov commented on include/picongpu/param/pml.param:
```diff
@@ -33,11 +33,11 @@ namespace fields
 namespace maxwellSolver
 {
-namespace yeePML
+namespace Pml
```
Is this spelling what you agreed upon with @psychocoderHPC ? To be personally it's the least logical of all, but I am fine with any, just wanted to clarify

steindev replied:
Yes, because it is supposed to become a type (struct/class) soon anyway and than the naming does not need to be changed again.

sbastrakov commented on share/picongpu/examples/FieldAbsorberTest/include/picongpu/param/pml.param:
```diff
@@ -33,7 +33,7 @@ namespace fields
+namespace Pml
 {
 
     /* The parameters in this file are only used if field solver is YeePML.
```
Perhaps you want to update this line as well, to be consistent with other changes in the comments?

steindev replied:
Actually not for the moment. As `YeePML` ist the only field solver that uses PMLs currently. I am just afraid that I/we will forget to change this line once there is another field solver making use of PMLs.
